modules/ding_reservation/ding_reservation.module

```diff
@@ -113,10 +113,19 @@ function ding_reservation_ding_entity_menu(&$items, $type, $path, $index) {
 }
 
 /**
- * AJAX callback used to check the reservability of the passed entity.
+ * AJAX callback used to check the reservability of the passed entity ids.
  */
-function ding_reservation_is_reservable_ajax_callback($entity) {
-  drupal_json_output(array('reservable' => $entity->is('reservable')));
+function ding_reservation_is_reservable_ajax_callback($local_ids) {
+  $local_ids = explode(',', $local_ids);
+  $result = [];
+
+  // The reservation provider decides reservability.
+  if (ding_provider_implements('reservation', 'is_reservable')) {
```

Comment:
As mentioned previously I think we can remove this check and simplify the code.

Reply:
I agree with you that it shouldn't be necessary.  

Just want to mention that ding_reservation uses this pattern in a lot of other places. Do you have any guidelines on when to use/not to use?

Reply:
Edit: After looking at code I see that the implements-pattern is only used when the provider callback is of optional nature.

Reply:
Fixed!